extensions/resteasy-reactive/jaxrs-client-reactive/deployment/src/main/java/io/quarkus/jaxrs/client/reactive/deployment/ClassRestClientContext.java

```diff
@@ -0,0 +1,127 @@
+package io.quarkus.jaxrs.client.reactive.deployment;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.MethodInfo;
+import org.objectweb.asm.Opcodes;
+
+import io.quarkus.deployment.GeneratedClassGizmoAdaptor;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.builditem.GeneratedClassBuildItem;
+import io.quarkus.gizmo.ClassCreator;
+import io.quarkus.gizmo.FieldDescriptor;
+import io.quarkus.gizmo.MethodCreator;
+import io.quarkus.gizmo.MethodDescriptor;
+import io.quarkus.gizmo.ResultHandle;
+
+class ClassRestClientContext implements AutoCloseable {
+
+    public final ClassCreator classCreator;
+    public final MethodCreator constructor;
+    public final MethodCreator clinit;
+
+    public final Map<Integer, FieldDescriptor> methodStaticFields = new HashMap<>();
+    public final Map<Integer, FieldDescriptor> methodParamAnnotationsStaticFields = new HashMap<>();
+    public final Map<Integer, FieldDescriptor> methodGenericParametersStaticFields = new HashMap<>();
+
+    public ClassRestClientContext(String name, BuildProducer<GeneratedClassBuildItem> generatedClasses,
+            String... interfaces) {
+        this(name, MethodDescriptor.ofConstructor(name), generatedClasses, Object.class, interfaces);
+    }
+
+    public ClassRestClientContext(String name, MethodDescriptor constructorDesc,
+            BuildProducer<GeneratedClassBuildItem> generatedClasses,
+            Class<?> superClass, String... interfaces) {
+
+        this.classCreator = new ClassCreator(new GeneratedClassGizmoAdaptor(generatedClasses, true),
+                name, null, superClass.getName(), interfaces);
+        this.constructor = classCreator.getMethodCreator(constructorDesc);
+        this.clinit = classCreator.getMethodCreator(MethodDescriptor.ofMethod(name, "<clinit>", void.class));
+        this.clinit.setModifiers(Opcodes.ACC_STATIC);
+    }
+
+    @Override
+    public void close() {
+        classCreator.close();
+    }
+
+    protected FieldDescriptor createJavaMethodField(ClassInfo interfaceClass, MethodInfo method, int methodIndex) {
```

Comment:
is there a case when we don't need a java method field?

Reply:
No, we always need it in:
- https://github.com/quarkusio/quarkus/pull/23090/files#diff-0c27cb2fbf4ead4ae377f712102826c9fbbe5ada510deda804e190838dd5b11fL46
- https://github.com/quarkusio/quarkus/pull/23090/files#diff-0c27cb2fbf4ead4ae377f712102826c9fbbe5ada510deda804e190838dd5b11fL69

Note that this is how was working before this PR, but this code was directly in JaxrsClientReactiveEnricher.